test/Main.hs

```diff
@@ -60,18 +61,18 @@ dotTest testname func genInc = testProperty testname $
 -- of the types
 --
 -- @
--- native :: (Int -> Vector Float -> Int -> Float )
--- fortran :: (Int -> Ptr Float -> Int -> IO Float )
+-- native :: (Int -> Vector Float -> Int -> a )
+-- fortran :: (Int -> Ptr Float -> Int -> IO a )
 -- @
 --
 -- produce byte equivalent to the results.   Vectors of length 1-10 are tested
 -- having elements that are in the range of approximately (epsilon/2,2/epsilon)
-ivifTest :: String
-        -> (Int -> V.Vector Float -> Int -> Float)
-        -> (Int -> Ptr Float -> Int -> IO Float)
+iviTest :: (Eq a, Show a) => String
```

Comment:
Polymorphized the datatype for the ivifTest harness so that it could be used for the `isamax` test